07-routing-gossip.md

```diff
@@ -268,8 +267,8 @@ The origin node:
   - SHOULD fill `addresses` with an address descriptor for each public network
   address that expects incoming connections.
   - MUST set `addrlen` to the number of bytes in `addresses`.
-  - MUST place non-zero typed address descriptors in ascending order.
-  - MAY place any number of zero-typed address descriptors anywhere.
+  - MUST place address descriptors in ascending order.
+  - SHOULD NOT place any zero-typed address descriptors anywhere.
```

Comment:
I think this 'SHOULD NOT' should be removed entirely.

Reply:
Heh, I like it for lack of ambiguity as you might otherwise interpret it as a legal thing to put a 0-type address descriptor and expect receivers to stop parsing the rest of the message (as they must stop parsing when they see the first undefined type).